### PR TITLE
grt: reInitTree when embedded routing has connectivity issues

### PR DESCRIPTION
**Pull request by eder-matheus**

### CODE REVIEW

**Comment by maliberty** [comment]
What are your results with this change?

**Comment by eder-matheus** [comment]
> What are your results with this change?

Both secure CI and public CI have no changes after these updates. For Seungwon's runs, my updates fix "Out of memory" errors by making additional checks during congestion iterations to early identify unroutable designs; and the invalid embedding (GRT-170 errors), by calling the "reInitTree" function when the wrong embedding is detected.

File: src/grt/src/fastroute/src/maze.cpp [hunk 1]
```diff
@@ -1005,7 +1006,15 @@ void FastRouteCore::updateRouteType1(const TreeNode* treenodes,
...
-    logger_->error(GRT, 169, "Invalid index for position ({}, {}).", E1x, E1y);
+    int x_pos = w_tile_ * (E1x + 0.5) + x_corner_;
+    int y_pos = h_tile_ * (E1y + 0.5) + y_corner_;
+    logger_->warn(GRT,
+                   169,
+                   "Net {}: Invalid index for position ({}, {}). Net degree: {}.",
+                   netName(nets_[net_id]),
+                   x_pos,
+                   y_pos,
+                   nets_[net_id]->numPins);
```
Comment (maliberty) [inline src/grt/src/fastroute/src/maze.cpp:1017]: Is it safe to continue after this warning with E1_pos == -1?

**Comment by maliberty** [comment]
> > What are your results with this change?
> 
> Both secure CI and public CI have no changes after these updates. For Seungwon's runs, my updates fix "Out of memory" errors by making additional checks during congestion iterations to early identify unroutable designs; and the invalid embedding (GRT-170 errors), by calling the "reInitTree" function when the wrong embedding is detected.

So no known autotuner issues exist?

File: src/grt/src/fastroute/src/maze.cpp (hunk 1 above)
Comment (eder-matheus) [reply src/grt/src/fastroute/src/maze.cpp:1017]: Yes, because the `checkRoute2DTree` function will detect the issue with the tree and will reset the tree.

**Comment by eder-matheus** [comment]
> > > What are your results with this change?
> > 
> > 
> > Both secure CI and public CI have no changes after these updates. For Seungwon's runs, my updates fix "Out of memory" errors by making additional checks during congestion iterations to early identify unroutable designs; and the invalid embedding (GRT-170 errors), by calling the "reInitTree" function when the wrong embedding is detected.
> 
> So no known autotuner issues exist?

Among the 28 cases sent by Seungwon, no.

**Review by maliberty — APPROVED** [review]

**Pull request merged**